File: packages/typescript/src/resources/export-import.ts
import { BaseResource } from './base.js';
import type { BundleImportResult, FullExportBundle } from '../types.js';

export class ExportImportResource extends BaseResource {
  exportAll(): Promise<FullExportBundle> {
    return this.client.request<FullExportBundle>({ path: 'export' });
  }

  export_all(): Promise<FullExportBundle> {
    return this.exportAll();
  }

  importAll(body: FullExportBundle): Promise<BundleImportResult> {
    return this.client.request<BundleImportResult, FullExportBundle>({
      path: 'import',
      method: 'POST',
      body,
    });
  }

  import_all(body: FullExportBundle): Promise<BundleImportResult> {
    return this.importAll(body);
  }
}

File: packages/flutter/lib/src/resources/export_import.dart
/// Export / Import resource — `/api/export` and `/api/import`.
library;

import '../models.dart';
import 'base.dart';

class ExportImportResource extends BaseResource {
  const ExportImportResource(super.client);

  Future<FullExportBundle> exportAll() async {
    final data = await client.request('GET', 'export');
    return FullExportBundle.fromJson(data as Map<String, dynamic>);
  }

  Future<BundleImportResult> importAll(FullExportBundle payload) async {
    final data = await client.request('POST', 'import', body: payload.toJson());
    return BundleImportResult.fromJson(data as Map<String, dynamic>);
  }
}
